backend/src/Designer/Infrastructure/GitRepository/AltinnAppGitRepository.cs

```diff
@@ -383,6 +384,62 @@ public string[] GetLayoutSetNames()
             return layoutSetNames;
         }
 
+        public bool EnsureValidFolderNameOfLayoutSetName(string layoutSetName, out string validFolderName)
```

Comment:
Method implementation should be changed. Look at CodeQL comments as well.
Also it'd be nice if we add some validation of layoutSetName on controller level. Regex for example.